Source/WebKit/WebProcess/WebCoreSupport/WebPermissionController.cpp

```diff
@@ -86,11 +89,23 @@ void WebPermissionController::removeObserver(WebCore::PermissionObserver& observ
     m_observers.remove(observer);
 }
 
-void WebPermissionController::permissionChanged(const WebCore::ClientOrigin& origin, const WebCore::PermissionDescriptor& descriptor, WebCore::PermissionState state)
+void WebPermissionController::permissionChanged(WebCore::PermissionName permissionName, const WebCore::SecurityOriginData& topOrigin)
 {
     for (auto& observer : m_observers) {
-        if (observer.origin() == origin && observer.descriptor() == descriptor)
-            observer.stateChanged(state);
+        if (observer.origin().topOrigin == topOrigin && observer.descriptor().name == permissionName) {
```

Comment:
Can we reverse the checks? The `observer.descriptor().name == permissionName` check is cheaper so it makes more sense to try this one first.

Reply:
Also, in WebKit we like early returns to reduce scoping.

This should be better as:
```
if (observer.descriptor().name != permissionName || observer.origin().topOrigin != topOrigin)
    continue;
```